src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+{
+    for (auto &thisHP : state.dataEIRFuelFiredHeatPump->heatPumps) {
+        if (!thisHP.companionCoilName.empty()) {
+            auto thisCoilName = UtilityRoutines::MakeUPPERCase(thisHP.name);
+            auto &thisCoilType = thisHP.EIRHPType;
+            auto targetCompanionName = UtilityRoutines::MakeUPPERCase(thisHP.companionCoilName);
+            for (auto &potentialCompanionCoil : state.dataEIRFuelFiredHeatPump->heatPumps) {
+                auto &potentialCompanionType = potentialCompanionCoil.EIRHPType;
+                auto potentialCompanionName = UtilityRoutines::MakeUPPERCase(potentialCompanionCoil.name);
+                if (potentialCompanionName == thisCoilName) {
+                    // skip the current coil
+                    continue;
+                }
+                if (potentialCompanionName == targetCompanionName) {
+                    if (thisCoilType == potentialCompanionType) {
+                        ShowSevereError(state,
+                                        "Invalid companion specification for EIR Plant Loop Fuel-Fired Heat Pump named \"" + thisCoilName + "\"");
+                        ShowContinueError(state, "For heating objects, the companion must be a cooling object, and vice-versa");
+                        ShowFatalError(state, "Invalid companion object causes program termination");
+                    }
+                    thisHP.companionHeatPumpCoil = &potentialCompanionCoil;
+                    break;
+                }
+            }
+            if (!thisHP.companionHeatPumpCoil) {
+                ShowSevereError(state, "Could not find matching companion heat pump coil.");
+                ShowContinueError(state, "Base coil: " + thisCoilName);
+                ShowContinueError(state, "Looking for companion coil named: " + targetCompanionName);
+                ShowFatalError(state, "Simulation aborts due to previous severe error");
+            }
+        }
+    }
+}
+
+void EIRFuelFiredHeatPump::processInputForEIRPLHP(EnergyPlusData &state)
+{
+    struct ClassType
+    {
+        DataPlant::PlantEquipmentType thisType;
+        std::string nodesType;
+        std::function<Real64(Real64, Real64)> calcLoadOutletTemp;
+        std::function<Real64(Real64, Real64)> calcQsource;
+        std::function<Real64(Real64, Real64)> calcSourceOutletTemp;
+
+        ClassType(DataPlant::PlantEquipmentType _thisType,
+                  std::string _nodesType,
+                  std::function<Real64(Real64, Real64)> _tLoadOutFunc,
+                  std::function<Real64(Real64, Real64)> _qSrcFunc,
+                  std::function<Real64(Real64, Real64)> _tSrcOutFunc)
+            : thisType(_thisType), nodesType(std::move(_nodesType)), calcLoadOutletTemp(std::move(_tLoadOutFunc)), calcQsource(std::move(_qSrcFunc)),
+              calcSourceOutletTemp(std::move(_tSrcOutFunc))
+        {
+        }
+    };
+    std::vector<ClassType> classesToInput = {
```

Comment:
This should be a `std::array`.

Reply:
ping @jcyuan2020

Reply:
In earlier trials, I tried to use `std::array` to replace `std::vector` (and also with `#include <array>` added at the top of the file), but the compiler would not let it pass. Not sure why `std::array` does not get compatible here?

Reply:
Thanks! That's actually it. Added the dimension it worked.
Changing the parent one to array as well.